# All-vs-All

```shell
DIR_BASE=/lizardfs/guarracino/wfmash-paper

FASTIX=/lizardfs/guarracino/git/fastix/target/release/fastix-331c1159ea16625ee79d1a82522e800c99206834
WFMASH=/lizardfs/guarracino/git/wfmash/build/bin/wfmash-251f4e1d7770723d3495e9d7b443f941798d4174
#WFMASH=/lizardfs/guarracino/git/wfmash/build/bin/wfmash-d7b696087f634f25e4b3de7dd521e1c4bfa3cf0e # to get alignment times in the PAF file
IMPG=/lizardfs/guarracino/git/impg/target/release/impg-54492fe21889ae686f09a7f8fad0e496914d9e82

conda create --prefix /lizardfs/guarracino/condatools/compleasm/0.2.5/ -c conda-forge -c bioconda compleasm=0.2.5 -y
```

```shell
sbatch -c 96 -p tux --job-name "primates16-serial" --wrap "cd /scratch; \time -v /lizardfs/guarracino/git/wfmash/build/bin/wfmash-042386f01e6a5fdee5bb7d325529f035fbad0a29 /lizardfs/guarracino/pggb-paper/assemblies/primates/primates16.20231205.fa.gz -t 96 -m > primates16.20231205.map.main-042386f.paf; mv primates16.20231205.map.main-042386f.paf /lizardfs/guarracino/wfmash-paper/"
sbatch -c 96 -p tux --job-name "primates16-parallel" --wrap "cd /scratch; \time -v /lizardfs/guarracino/git/forks/wfmash/build/bin/wfmash-parallel-indexing /lizardfs/guarracino/pggb-paper/assemblies/primates/primates16.20231205.fa.gz -t 96 -m > primates16.20231205.parallel-indexing.paf; mv primates16.20231205.parallel-indexing.paf /lizardfs/guarracino/wfmash-paper/"

/lizardfs/guarracino/git/forks/wfmash/build/bin/wfmash-parallel-indexing /lizardfs/guarracino/pggb-paper/assemblies/primates/primates16.20231205.fa.gz -t 48 -m > primates16.20231205.map.main.paf
```

## Data

```shell
mkdir -p $DIR_BASE/assemblies/scerevisiae
cd $DIR_BASE/assemblies/scerevisiae

wget -c https://raw.githubusercontent.com/pangenome/pggb/master/docs/data/scerevisiae.yprp.urls
cat scerevisiae.yprp.urls | grep mt -v | parallel -j 4 'wget -q {} && echo got {}'

# PanSN
ls *.fa.gz | while read FASTA; do
    SAMPLE=$(echo $FASTA | cut -f 1 -d '.');
    echo $SAMPLE

    zcat $FASTA | sed "s/^>/>$SAMPLE#1#/g" | bgzip -@ 48 -l 9 > $SAMPLE.fasta.gz && samtools faidx $SAMPLE.fasta.gz
done
rm *.fa.gz

zcat *.fasta.gz | bgzip -@ 48 -l 9 > scerevisiae7.fa.gz && samtools faidx scerevisiae7.fa.gz
```

## BUSCO genes

### Yeast

```shell
mkdir -p $DIR_BASE/busco
cd $DIR_BASE/busco

conda activate /lizardfs/guarracino/condatools/compleasm/0.2.5 

compleasm download -L $DIR_BASE/busco/databases saccharomycetes

mkdir -p $DIR_BASE/busco/scerevisiae
cd $DIR_BASE/busco/scerevisiae
ls $DIR_BASE/assemblies/scerevisiae/*fasta.gz | while read FASTA; do
    SAMPLE=$(basename $FASTA .fasta.gz)
    echo $SAMPLE
    
    DIR_OUTPUT=$DIR_BASE/busco/scerevisiae/$SAMPLE
    mkdir -p $DIR_OUTPUT
    sbatch -c 48 -p workers  --job-name busco-saccharomycetes-$SAMPLE --wrap "hostname; cd /scratch; compleasm run -a $FASTA -o $DIR_OUTPUT -t 48 -l saccharomycetes -L $DIR_BASE/busco/databases; mv $DIR_OUTPUT/summary.txt $DIR_OUTPUT/$SAMPLE.summary.saccharomycetes.txt"
done

conda deactivate

# Table to BED
ls $DIR_BASE/assemblies/scerevisiae/*fasta.gz | while read FASTA; do
    SAMPLE=$(basename $FASTA .fasta.gz)
    echo $SAMPLE
    
    DIR_OUTPUT=$DIR_BASE/busco/scerevisiae/$SAMPLE
    grep Missing -v $DIR_OUTPUT/saccharomycetes_odb10/full_table.tsv | sed '1d' | awk -v OFS='\t' '{print($3,$4,$5,$1,$7,$6,$2)}' > $DIR_OUTPUT/saccharomycetes_odb10/$SAMPLE.busco-genes.bed
done
```

### Primates

```shell
mkdir -p $DIR_BASE/busco/databases
cd $DIR_BASE/busco

conda activate /lizardfs/guarracino/condatools/compleasm/0.2.5

compleasm download -L $DIR_BASE/busco/databases primates

mkdir -p $DIR_BASE/busco/primates
cd $DIR_BASE/busco/primates
cut -f 1 $PATH_PRIMATES16_FASTA.fai | cut -f 1,2 -d '#' | sort | uniq | while read SAMPLE; do
    echo $SAMPLE
    mkdir -p $DIR_BASE/busco/$SAMPLE
    samtools faidx $PATH_PRIMATES16_FASTA $(grep "^$SAMPLE" $PATH_PRIMATES16_FASTA.fai | cut -f 1) | bgzip -@ 48 -l 9 > $DIR_BASE/busco/$SAMPLE/$SAMPLE.fa.gz
    sbatch -c 48 -p workers  --job-name busco-primates-$SAMPLE --wrap "hostname; cd /scratch; compleasm run -a $DIR_BASE/busco/$SAMPLE/$SAMPLE.fa.gz -o $DIR_BASE/busco/$SAMPLE -t 48 -l primates -L $DIR_BASE/busco/databases; mv $DIR_BASE/busco/$SAMPLE/summary.txt $DIR_BASE/busco/$SAMPLE/summary.primates.$SAMPLE.txt; rm $DIR_BASE/busco/$SAMPLE/$SAMPLE.fa.gz"
done
conda deactivate

mkdir -p $DIR_BASE/busco/tmp
cp $DIR_BASE/busco/*/summary.primates.*.txt $DIR_BASE/busco/tmp
# Script adapted from https://gitlab.com/ezlab/busco/-/raw/master/scripts/generate_plot.py
python3 /lizardfs/guarracino/ratty/scripts/generate_plot_for_mini_busco.py -wd $DIR_BASE/busco/tmp
mv $DIR_BASE/busco/tmp/busco_figure.png $DIR_BASE/busco/busco_figure.primates16.primates.png
mv $DIR_BASE/busco/tmp/busco_figure.R $DIR_BASE/busco/busco_figure.primates16.primates.R
rm -rf $DIR_BASE/busco/tmp

cut -f 1 $PATH_PRIMATES16_FASTA.fai | cut -f 1,2 -d '#' | sort | uniq | while read SAMPLE; do
    echo $SAMPLE
    grep Missing -v $DIR_BASE/busco/$SAMPLE/primates_odb10/full_table.tsv | sed '1d' | awk -v OFS='\t' '{print($3,$4,$5,$1,$7,$6,$2)}' > $DIR_BASE/busco/$SAMPLE/primates_odb10/$SAMPLE.busco-genes.bed
done
```

## Alignment

### Yeast

```shell
FASTA=$DIR_BASE/assemblies/scerevisiae/scerevisiae7.fa.gz
NAME=$(basename $FASTA .fa.gz)
DIR_OUTPUT=$DIR_BASE/alignment/$NAME

mkdir -p $DIR_OUTPUT
cd $DIR_OUTPUT

s=5k
p=90
n=1
c=20k

cut -f 1,2 $FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    sbatch -c 48 -p allnodes --job-name "$NAME-vs-$TARGET-mapping" --wrap "hostname; \
        $WFMASH -t 48 \
        -P $PREFIX \
        -Y '#' \
        -n $n \
        -p $p \
        -s $s \
        -c $c \
        $FASTA \
        -m > $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.map.paf \
        2> $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.map.log"
done

cut -f 1,2 $FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    sbatch -c 48 -p allnodes --job-name "$NAME-vs-$TARGET-mapping" --wrap "hostname; \
        $WFMASH -t 48 \
        -P $PREFIX \
        -Y '#' \
        -n $n \
        -p $p \
        -s $s \
        -c $c \
        $FASTA \
        -i $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.map.paf \
        > $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.aln.paf \
        2> $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.aln.log"
done
```

### Primates

Mapping:

```shell
WFMASH=/home/guarracino/tools/wfmash/build/bin/wfmash-d7b696087f634f25e4b3de7dd521e1c4bfa3cf0e # to get alignment times in the PAF file

DIR_OUT=$DIR_BASE/mapping
mkdir -p $DIR_OUT
cd $DIR_OUT

cut -f 1,2 $PATH_PRIMATES16_FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    for p in 70; do
        sbatch -c 48 -p allnodes -x octopus05 --job-name "primates-vs-$TARGET-p$p-mapping" --wrap "hostname; \
            \time -v $WFMASH -t 48 \
            -m \
            -P $PREFIX \
            --one-to-one \
            -Y '#' \
            -n 1 \
            -p $p \
            -s 5k \
            -c 20k \
            $PATH_PRIMATES16_FASTA \
            > $DIR_OUT/$TARGET.p$p.map.paf \
            2> $DIR_OUT/$TARGET.p$p.map.log"
    done
done
```

Alignment:

```shell
WFMASH=/home/guarracino/tools/wfmash/build/bin/wfmash-d7b696087f634f25e4b3de7dd521e1c4bfa3cf0e # to get alignment times in the PAF file

DIR_OUT=$DIR_BASE/alignment
mkdir -p $DIR_OUT
cd $DIR_OUT

cut -f 1,2 $PATH_PRIMATES16_FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    for p in 70; do
        sbatch -c 48 -p tux --job-name "primates-vs-$TARGET-p$p-alignment" --wrap "hostname; \
            \time -v $WFMASH -t 48 \
            -i $DIR_BASE/mapping/$TARGET.p$p.map.paf \
            -P $PREFIX \
            --one-to-one \
            -Y '#' \
            -n 1 \
            -p $p \
            -s 5k \
            -c 20k \
            $PATH_PRIMATES16_FASTA \
            > $DIR_OUT/$TARGET.p$p.aln.paf \
            2> $DIR_OUT/$TARGET.p$p.aln.log"
    done
done
```

### HPRCy1

Prepare:

```shell
cat /lizardfs/erikg/HPRC/year1v2genbank/assemblies/*v2_genbank.fa <(zcat /lizardfs/erikg/HPRC/year1v2genbank/assemblies/HG02080.paternal.f1_assembly_v2_genbank.fa.gz) <(zcat /lizardfs/guarracino/pggb-paper/assemblies/primates/primates16.20231205.fa.gz) | bgzip -l 9 -@ 128 > /lizardfs/guarracino/HPRC/HPRCy1+primates16.fa.gz && samtools faidx /lizardfs/guarracino/HPRC/HPRCy1+primates16.fa.gz
```

Mapping:

```shell
cd $DIR_BASE/mapping

cut -f 1,2 $PATH_PRIMATES16_FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    for p in 70; do
        sbatch -c 96 -p tux --job-name "HPRCy1-vs-$TARGET-p$p-mapping" --wrap "hostname; \
            \time -v $WFMASH -t 96 \
            -m \
            -P $PREFIX \
            --one-to-one \
            -Y '#' \
            -n 1 \
            -p $p \
            -s 5k \
            -c 20k \
            /lizardfs/guarracino/HPRC/HPRCy1+primates16.fa.gz \
            > $DIR_BASE/mapping/HPRCy1-vs-$TARGET.p$p.map.paf \
            2> $DIR_BASE/mapping/HPRCy1-vs-$TARGET.p$p.map.log"
    done
done
```

Alignment:

```shell
WFMASH=/home/guarracino/tools/wfmash/build/bin/wfmash-d7b696087f634f25e4b3de7dd521e1c4bfa3cf0e # to get alignment times in the PAF file

DIR_BASE=/lizardfs/guarracino/primates
PATH_PRIMATES16_FASTA=/lizardfs/guarracino/pggb-paper/assemblies/primates/primates16.20231205.fa.gz
cd $DIR_BASE/alignment

cut -f 1,2 $PATH_PRIMATES16_FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    for p in 70; do
        sbatch -c 96 -p tux --job-name "HPRCy1-vs-$TARGET-p$p-alignment" --wrap "hostname; \
            \time -v $WFMASH -t 96 \
            -i $DIR_BASE/mapping/HPRCy1-vs-$TARGET.p$p.map.paf \
            -P $PREFIX \
            --one-to-one \
            -Y '#' \
            -n 1 \
            -p $p \
            -s 5k \
            -c 20k \
            /lizardfs/guarracino/HPRC/HPRCy1+primates16.fa.gz \
            > $DIR_BASE/alignment/HPRCy1-vs-$TARGET.p$p.aln.paf \
            2> $DIR_BASE/alignment/HPRCy1-vs-$TARGET.p$p.aln.log"
    done
done
```


## Evaluation

```shell
# # Number of GENCODE genes in the target
# total_genes_in_target=$(cat $path_gencode_genes_target | wc -l)

# #cat alignment/$prefix.paf | awk -v OFS='\t' '{print $1, $3, $4, "", "", $5}' >alignment/$prefix.$query.bed
# cat alignment/$prefix.paf | awk -v OFS='\t' '{print $6, $8, $9, "", "", "+"}' | sed 's/grch38#//g' >alignment/$prefix.$target.bed

# # Number of GENCODE genes not entirely covered
# missing_genes_in_target=$(bedtools subtract -a $path_gencode_genes_target -b alignment/$prefix.$target.bed | cut -f 4 | sort | uniq | wc -l)

# missing_gene_ratio_target=$(echo "scale=4; 1 - $missing_genes_in_target / $total_genes_in_target" | bc)

# echo $query $target $total_genes_in_target $s $l $p $n $w $missing_gene_ratio_target $missing_genes_in_target | tr ' ' '\t' >>gencode_evaluation.mapping.tsv
```

### Yeast

```shell
FASTA=$DIR_BASE/assemblies/scerevisiae/scerevisiae7.fa.gz
NAME=$(basename $FASTA .fa.gz)
DIR_OUTPUT=$DIR_BASE/alignment/scerevisiae7

# All single BUSCO genes
cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep Single > $DIR_OUTPUT/scerevisiae.busco-genes.single.bed

s=5k
p=90
n=1
c=20k

# All-vs-1
cut -f 1,2 $FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    PAF=$DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.aln.paf

    cut -f 4 $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | sort | uniq | while read GENE; do
        #echo $GENE
        

        ###################### TO DO #####################################################################
        ###################### TO DO #####################################################################
        #check the sizes of this gene in all genomes
        #take the max
        #compute max-len for bedtools slop
        ###################### TO DO #####################################################################
        ###################### TO DO #####################################################################


        # compleasm-regions minus impg-regions
        bedtools subtract \
            -a <(cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep $GENE -w | bedtools sort) \
            -b <($IMPG -p $PAF -b <(grep "^$TARGET" $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | grep $GENE -w | bedtools slop -b 0 -g <(cut -f 1,2 $FASTA.fai)) | bedtools sort)
        #bedtools subtract \
        #    -a <(grep $GENE -w $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | bedtools sort) \
        #    -b <($IMPG -p $PAF -b <(grep "^$TARGET" $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | grep $GENE -w | bedtools slop #-b 0 -g <(cut -f 1,2 $FASTA.fai)) | bedtools sort)

        # # It computes both the depth and breadth of coverage of features in file B on the features in file A.
        # bedtools coverage \
        #     -a <(cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep $GENE -w | bedtools sort) \
        #     -b <($IMPG -p $PAF -b <(cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep "^$TARGET" | grep $GENE -w) | bedtools sort) | column -t
        #cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep "^$TARGET" | grep $GENE -w
        #UNCOVERED_BP=$(awk 'BEGIN{sum=0}{sum += $3 - $2} END {print sum}' $GENE.uncovered.bed)
    done > $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.uncovered.bed
done

# Number of evaluated genes
cat $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | cut -f 4 | sort | uniq | wc -l

cut -f 1,2 $FASTA.fai -d '#' | sort | uniq | while read TARGET; do
    PREFIX="$TARGET#"
    echo $PREFIX

    # Number of genes with missing bases
    cut -f 4 $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.uncovered.bed | sort | uniq | wc -l

    # Amount of uncovered gene bases
    awk 'BEGIN{sum=0}{sum += $3 - $2} END {print sum}' $DIR_OUTPUT/$NAME-vs-$TARGET.s$s.p$p.n$n.c$c.uncovered.bed
done









# All-vs-all
PAF=$DIR_OUTPUT/$NAME.s$s.p$p.n$n.c$c.aln.paf
cat $DIR_OUTPUT/$NAME-vs-*.s$s.p$p.n$n.c$c.aln.paf > $DIR_OUTPUT/$NAME.s$s.p$p.n$n.c$c.aln.paf

cut -f 4 $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | sort | uniq | while read GENE; do
    #echo $GENE

    # impg-regions - compleasm-regions
    bedtools subtract \
        -a <(cat $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | grep $GENE -w | bedtools sort) \
        -b <($IMPG -p $PAF -b <(grep "^$TARGET" $DIR_OUTPUT/scerevisiae.busco-genes.single.bed | grep $GENE -w) | bedtools sort)

    # # It computes both the depth and breadth of coverage of features in file B on the features in file A.
    # bedtools coverage \
    #     -a <(cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep $GENE -w | bedtools sort) \
    #     -b <($IMPG -p $PAF -b <(cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep "^$TARGET" | grep $GENE -w) | bedtools sort) | column -t
    #cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep "^$TARGET" | grep $GENE -w
    #UNCOVERED_BP=$(awk 'BEGIN{sum=0}{sum += $3 - $2} END {print sum}' $GENE.uncovered.bed)
done > $DIR_OUTPUT/$NAME.s$s.p$p.n$n.c$c.uncovered.bed

# Number of evaluated genes
cat $DIR_BASE/busco/scerevisiae/*/*/*.busco-genes.bed | grep Single | cut -f 4 | sort | uniq | wc -l

# Number of genes with missing bases
cut -f 4 $DIR_OUTPUT/$NAME.s$s.p$p.n$n.c$c.uncovered.bed | sort | uniq | wc -l

# Amount of uncovered gene bases
awk 'BEGIN{sum=0}{sum += $3 - $2} END {print sum}' $DIR_OUTPUT/$NAME.s$s.p$p.n$n.c$c.uncovered.bed


```

### Primates

```shell
cat $DIR_BASE/busco/primates/*/primates_odb10/*.busco-genes.bed | grep Single > $DIR_BASE/busco/primates/primates16.busco-genes.bed

SAMPLE=all
bedtools sort -i $DIR_BASE/busco/primates/primates16.busco-genes.bed | bedtools merge > $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.bed
bedtools sort -i $DIR_BASE/busco/primates/primates16.busco-genes.bed | bedtools merge > $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.bed

cat /lizardfs/erikg/primates/primates16.20231205_wfmash-v0.12.5/*.map.paf | awk -v OFS='\t' '{print $1, $3, $4, "", "", $5}'  | bedtools sort | bedtools merge > $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.map.bed
cat /lizardfs/erikg/primates/primates16.20231205_wfmash-v0.12.5/*.aln.paf | awk -v OFS='\t' '{print $1, $3, $4, "", "", $5}'  | bedtools sort | bedtools merge > $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.alg.bed
cat /lizardfs/erikg/primates/primates16.20231205_wfmash-v0.12.5/*.map.paf | awk -v OFS='\t' '{print $6, $8, $9, "", "", "+"}' | bedtools sort | bedtools merge > $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.map.bed
cat /lizardfs/erikg/primates/primates16.20231205_wfmash-v0.12.5/*.aln.paf | awk -v OFS='\t' '{print $6, $8, $9, "", "", "+"}' | bedtools sort | bedtools merge > $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.alg.bed

ALL_QUERY=$(cat $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.bed | awk 'BEGIN{sum=0}{sum+=$3-$2}END{print(sum)}')
MAP_QUERY=$(bedtools intersect -a $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.bed -b $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.map.bed  | awk 'BEGIN{sum=0}{sum+=$3-$2}END{print(sum)}')
ALG_QUERY=$(bedtools intersect -a $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.bed -b $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.query.alg.bed  | awk 'BEGIN{sum=0}{sum+=$3-$2}END{print(sum)}')
echo "Map-Query: $(echo "scale=8; $MAP_QUERY / $ALL_QUERY" | bc)"
echo "Alg-Query: $(echo "scale=8; $ALG_QUERY / $ALL_QUERY" | bc)"

ALL_TARGET=$(cat $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.bed | awk 'BEGIN{sum=0}{sum+=$3-$2}END{print(sum)}')
MAP_TARGET=$(bedtools intersect -a $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.bed -b $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.map.bed  | awk 'BEGIN{sum=0}{sum+=$3-$2}END{print(sum)}')
ALG_TARGET=$(bedtools intersect -a $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.bed -b $DIR_BASE/busco/primates/primates16.busco-genes.$SAMPLE.target.alg.bed  | awk 'BEGIN{sum=0}{sum+=$3-$2}END{print(sum)}')
echo "Map-Target: $(echo "scale=8; $MAP_TARGET / $ALL_TARGET" | bc)"
echo "Alg-Target: $(echo "scale=8; $ALG_TARGET / $ALL_TARGET" | bc)"
```

Compute total runtime and memory usage:

```shell
#!/bin/bash

# Initialize total seconds and max memory usage counters
totalSeconds=0
maxMemory=0

# Process each line that contains elapsed time
grep elapsed *.log | grep 'wfmash\|mashmap' -v | while IFS= read -r line; do
    # Extract the elapsed time and convert it to seconds
    elapsed=$(echo "$line" | awk -F'elapsed' '{print $1}' | awk '{print $NF}')
    
    # Check if the elapsed time includes hours
    if [[ $elapsed == *:*:* ]]; then
        hours=$(echo "$elapsed" | cut -d':' -f1)
        minutes=$(echo "$elapsed" | cut -d':' -f2)
        seconds=$(echo "$elapsed" | cut -d':' -f3)
    else
        hours=0
        minutes=$(echo "$elapsed" | cut -d':' -f1)
        seconds=$(echo "$elapsed" | cut -d':' -f2)
    fi
    
    # Convert to total seconds, accounting for potential milliseconds
    total=$(echo "$hours*3600 + $minutes*60 + $seconds" | bc -l)
    
    # Accumulate the total
    totalSeconds=$(echo "$totalSeconds + $total" | bc)
    
    # Extract max memory usage and compare/update
    memory=$(echo "$line" | awk '{print $(NF)}' | sed 's/maxresident)k//')

    if (( $(echo "$memory > $maxMemory" | bc -l) )); then
        maxMemory=$memory
    fi
done

# Since we're dealing with potentially floating point numbers for seconds, round the total seconds
totalSeconds=$(printf "%.0f" "$totalSeconds")

# Convert total seconds back to HH:MM:SS format
hours=$(echo "$totalSeconds/3600" | bc)
minutes=$(echo "($totalSeconds%3600)/60" | bc)
seconds=$(echo "$totalSeconds%60" | bc)

# Convert max memory from kilobytes to gigabytes
maxMemoryGB=$(echo "scale=2; $maxMemory/1024/1024" | bc)

# Format the total time as HH:MM:SS and print max memory usage
printf "Total elapsed time: %02d:%02d:%02d\n" $hours $minutes $seconds
printf "Maximum memory usage: %.2fGB\n" $maxMemoryGB
```

# Sequence complexity

```shell
/lizardfs/guarracino/git/sattools/target/release/sattools /lizardfs/guarracino/robertsonian_translocation/assemblies/chm13v2.0.fa -k 19 -o chm13v2.sattols.bed -t 48

/lizardfs/guarracino/git/sattools/target/release/sattools <(zcat /lizardfs/guarracino/pggb-paper/assemblies/primates/hg002v101.fa.gz) -k 19 -o hg002v101.sattols.bed -t 48
```


### TESTS

```shell
mkdir -p $DIR_BASE/alignment/scerevisiae
cd $DIR_BASE/alignment/scerevisiae

#### FOR DEVELOPMENT############################################################
samtools faidx $DIR_BASE/assemblies/scerevisiae/S288C.fasta.gz $(head -n 1 $DIR_BASE/assemblies/scerevisiae/S288C.fasta.gz.fai | cut -f 1) | bgzip -@ 48 -l 9 > S288C_1_chrI.fa.gz && samtools faidx S288C_1_chrI.fa.gz
#samtools faidx $DIR_BASE/assemblies/scerevisiae/DBVPG6044.fasta.gz $(head -n 1 $DIR_BASE/assemblies/scerevisiae/DBVPG6044.fasta.gz.fai | cut -f 1) -i | bgzip -@ 48 -l 9 > DBVPG6044_1_chrI.fa.gz && samtools faidx DBVPG6044_1_chrI.fa.gz
read CHR LENGTH <<< $(head -n 1 $DIR_BASE/assemblies/scerevisiae/DBVPG6044.fasta.gz.fai | cut -f 1,2)
# Calculate the midpoint of the chromosome (assuming integer division truncates the fraction)
MIDPOINT=$((LENGTH / 2))
# Extract and reverse complement the first half of the chromosome
cat <(samtools faidx $DIR_BASE/assemblies/scerevisiae/DBVPG6044.fasta.gz $CHR:1-$MIDPOINT -i) <(samtools faidx $DIR_BASE/assemblies/scerevisiae/DBVPG6044.fasta.gz $CHR:$((MIDPOINT + 1))-$LENGTH) | bgzip -@ 48 -l 9 > DBVPG6044_1_chrI.fa.gz && samtools faidx DBVPG6044_1_chrI.fa.gz

PATH_QUERY_BED=$DIR_BASE/busco/scerevisiae/DBVPG6044/saccharomycetes_odb10/DBVPG6044.busco-genes.bed
PATH_TARGET_BED=$DIR_BASE/busco/scerevisiae/S288C/saccharomycetes_odb10/S288C.busco-genes.bed

PATH_PAF=$DIR_BASE/alignment/scerevisiae/DBVPG6044-vs-S288C.paf
$WFMASH $DIR_BASE/assemblies/scerevisiae/S288C.fasta.gz $DIR_BASE/assemblies/scerevisiae/DBVPG6044.fasta.gz -t 48 > $PATH_PAF

#PATH_PAF=$DIR_BASE/alignment/scerevisiae/DBVPG6044-vs-S288C.chrI.paf
#$WFMASH S288C_1_chrI.fa.gz DBVPG6044_1_chrI.fa.gz -t 48 -s 100 -p 50 > $PATH_PAF
#PATH_QUERY_BED=/lizardfs/guarracino/wfmash-paper/alignment/scerevisiae/busco_inverted_chr/saccharomycetes_odb10/DBVPG6044_1_chrI.rev.bed
################################################################################

PATH_PAF=$DIR_BASE/alignment/scerevisiae/scerevisiae7.paf
$WFMASH $DIR_BASE/assemblies/scerevisiae/scerevisiae7.fa.gz -t 48 --lower-triangular > $PATH_PAF

PATH_BUSCO_BED=$DIR_BASE/alignment/scerevisiae/scerevisiae.busco-genes.single.bed
cat $DIR_BASE/busco/scerevisiae/*/saccharomycetes_odb10/*.busco-genes.bed | grep 'Single$' > $PATH_BUSCO_BED # Put all the BUSCO genes in a single file


bash $DIR_BASE/scripts/check_features/check_features.sh $PATH_PAF $PATH_BUSCO_BED 59 ./scerevisiae7.m50 /scratch

\time -v $WFMASH $DIR_BASE/assemblies/scerevisiae/S288C.fasta.gz $DIR_BASE/assemblies/scerevisiae/scerevisiae7.fa.gz -t 48 -p 99   > $DIR_BASE/alignment/scerevisiae/scerevisiae7.wfm-p99.paf
\time -v minimap2 $DIR_BASE/assemblies/scerevisiae/S288C.fasta.gz $DIR_BASE/assemblies/scerevisiae/scerevisiae7.fa.gz -t 48 -x asm5 --eqx -c > $DIR_BASE/alignment/scerevisiae/scerevisiae7.mm2-asm5.paf

bash $DIR_BASE/scripts/check_features/check_features.sh $DIR_BASE/alignment/scerevisiae/scerevisiae7.wfm-p99.paf  $PATH_BUSCO_BED 50 ./scerevisiae7.m50.wfm 48 /scratch
bash $DIR_BASE/scripts/check_features/check_features.sh $DIR_BASE/alignment/scerevisiae/scerevisiae7.mm2-asm5.paf $PATH_BUSCO_BED 50 ./scerevisiae7.m50.mm2 48 /scratch

############################################################################################################################################
# Working implementation
########################
# NAME=$(basename $PAF .paf)
# bedtools intersect -a <(awk -v OFS='\t' '{print($1,$3,$4,$0)}' $PAF) -b <(grep 'Single$' $PATH_QUERY_BED)  -wa -wb | cut -f 4- > $NAME.query.paf
# bedtools intersect -a <(awk -v OFS='\t' '{print($6,$8,$9,$0)}' $PAF) -b <(grep 'Single$' $PATH_TARGET_BED) -wa -wb | cut -f 4- > $NAME.target.paf

# # Join the two files
# join -1 1 -2 1 \
#     <(awk -v OFS='\t' '{concat=$1; for(i=2;i<=20;i++) concat=concat "___" $i; concat=concat "___" $24; print(concat,$21,$22,$23,$25,$26,$27)}' $NAME.query.paf  | sort -T /scratch) \
#     <(awk -v OFS='\t' '{concat=$1; for(i=2;i<=20;i++) concat=concat "___" $i; concat=concat "___" $24; print(concat,$21,$22,$23,$25,$26,$27)}' $NAME.target.paf | sort -T /scratch) |
#     # Deconcatenate the joined column back into original columns
#     # What happens if the gene is covered by more mappings in one genome than in the other???????????????????????????????????
#     awk '{
#         # Split the first field into array `a` using the "|" delimiter
#         n=split($1, a, "___");
#         # Reconstruct the original columns
#         for(i=1; i<=20; i++) printf("%s\t", a[i]);
#         printf("%s\t%s\t%s\t%s\t%s\t%s\t%s\t", $2, $3, $4, a[21], $5, $6, $7);
#         printf("%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $8, $9, $10, a[21], $11, $12, $13);
#     }' | cut -f 1-12,20- | pigz -9 > $NAME.joined.paf.gz
# # Number of BUSCO genes not covered by the mapping on both genomes
# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
# zcat $NAME.joined.paf.gz | python3 x.py > alignments-checked.tsv
############################################################################################################################################


# Check with SafFire
GENE=2456at4891
zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | cut -f 1-13 > $GENE.paf
zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | rev | cut -f 8-14 | rev > $GENE.in-query.bed
zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | rev | cut -f 1-7  | rev > $GENE.in-target.bed
~/git/rustybam/target/release/rb stats --paf $GENE.paf > $GENE.saffire
~/git/rustybam/target/release/rb break-paf --max-size 1 $GENE.paf | ~/git/rustybam/target/release/rb stats --paf > $GENE.saffire
~/git/rustybam/target/release/rb liftover <(zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | cut -f 1-13) --bed <(zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | rev | cut -f 1-7 | rev)
~/git/rustybam/target/release/rb liftover <(zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | cut -f 1-13) --bed <(zgrep $GENE DBVPG6044-vs-S288C.chrI.joined.paf.gz | rev | cut -f 8-14| rev) --qbed

############################################################################################################################################
# Slower, but more controllable implementation
##############################################
# # Common single BUSCO genes
# (grep 'Single$' $PATH_QUERY_BED | cut -f 4 | sort; grep 'Single$' $PATH_TARGET_BED | cut -f 4 | sort) | sort | uniq -d > common.busco-genes.txt

# # For each gene, get the mappings that cover them
# # Ignore the strand because it depends on the gene too. For example, 0at4891 is on - for both genomes, and query and target are aligned with the same orientation, so no problem
# cat common.busco-genes.txt | while read GENE; do
#     #echo $GENE >&2

#     # CAN WE GENERATE THE INTERSECTION FOR ALL GENES AT ONCE?????????????????????????????????????????????????????????????????????????????????
#     grep $GENE $PATH_QUERY_BED -w | bedtools intersect -a <(awk -v OFS='\t' '{print($1,$3,$4,$0)}' $PAF) -b - -wa -wb | cut -f 4- | sort > $GENE.query.paf
#     grep $GENE $PATH_TARGET_BED -w | bedtools intersect -a <(awk -v OFS='\t' '{print($6,$8,$9,$0)}' $PAF) -b - -wa -wb | cut -f 4- | sort > $GENE.target.paf

#     # should_continue=false
#     # if [ ! -s "$GENE.query.paf" ]; then
#     #     echo "Warning: $GENE.query.paf is empty" >&2
#     #     should_continue=true
#     # fi
#     # if [ ! -s "$GENE.target.paf" ]; then
#     #     echo "Warning: $GENE.target.paf is empty" >&2
#     #     should_continue=true
#     # fi
#     # if [ "$should_continue" = true ]; then
#     #     continue
#     # fi
    
#     # Check if either file has more than one row and emit a warning if so
#     if [ $(wc -l < "$GENE.query.paf") -gt 1 ] || [ $(wc -l < "$GENE.target.paf") -gt 1 ]; then
#         echo "Warning: More than one mapping found for $GENE, skipping..." >&2
#         continue # Skip to the next GENE
#     fi

#     # What happens if the gene is covered by more mappings in one genome than in the other?
#     python -c "gene=\"$GENE\"; f1=open(f'{gene}.query.paf'); f2=open(f'{gene}.target.paf'); d1={}; d2={}; [d1.setdefault(' '.join(line.split()[:-7]), []).append(line.split()[-7:]) for line in f1]; [d2.setdefault(' '.join(line.split()[:-7]), []).append(line.split()[-7:]) for line in f2]; [print(k, *v[0], *d2[k][0]) for k, v in d1.items() if k in d2]; f1.close(); f2.close()" | cut -f 1-12,20- -d ' ' | tr ' ' '\t'
    
#     rm  $GENE.query.paf $GENE.target.paf
# done | pigz -9 > alignments-to-check.paf.gz

# # Number of BUSCO genes not covered by the mapping on both genomes
# comm -23 common.busco-genes.txt <(zcat alignments-to-check.paf.gz | rev | cut -f 4 | rev) | wc -l

# zcat alignments-to-check.paf.gz | python3 x.py > alignments-checked.tsv
############################################################################################################################################



~/git/rustybam/target/release/rb break-paf --max-size 1 $GENE.paf | ~/git/rustybam/target/release/rb stats --paf > $GENE.saffire












cat $DIR_BASE/busco*/primates_odb10/*.busco-genes.bed | grep Single > $DIR_BASE/busco/primates16.busco-genes.bed

grep 18084at9443 $DIR_BASE/busco/primates16.busco-genes.bed > $DIR_BASE/busco/primates16.busco-genes.18084at9443.bed
SAMPLE=grch38#1
PAF=/lizardfs/erikg/primates/primates16.20231205_wfmash-v0.12.5/$SAMPLE.aln.paf
grep "grch38#1#chr13" $PAF | grep chm13 > $SAMPLE.aln.chr13.paf
bash /lizardfs/guarracino/wfmash-paper/scripts/check_features/check_features.sh $SAMPLE.aln.chr13.paf $DIR_BASE/busco/primates16.busco-genes.18084at9443.bed 50 $DIR_BASE/$SAMPLE.aln.chr13.m50.18084at9443 48 /scratch

seq 6 9 | while read f; do
    sbatch -c 48 -p allnodes -w tux0$f --wrap "hostname; cd /scratch/ && rm sort* *paf *paf.gz *bed"
done

cd $DIR_BASE
cut -f 1 $PATH_PRIMATES16_FASTA.fai | cut -f 1,2 -d '#' | sort | uniq | while read SAMPLE; do
    PAF=/lizardfs/erikg/primates/primates16.20231205_wfmash-v0.12.5/$SAMPLE.aln.paf

    sbatch -c 48 -p allnodes --job-name check-busco-$SAMPLE --wrap "hostname; \time -v bash /lizardfs/guarracino/wfmash-paper/scripts/check_features/check_features.sh $PAF $DIR_BASE/busco/primates16.busco-genes.bed 50 $DIR_BASE/$SAMPLE.aln.m50 48 /scratch"
done
```
