# All-vs-one

## Assembly vs reference

### Installation

```bash
conda create --prefix /lizardfs/guarracino/condatools/wfmash-paper/ -c conda-forge -c bioconda anchorwave=1.2.2 minimap2=2.26 bcftools=1.19 samtools=1.19.2 syri=1.6.3 mummer4=4.0.0rc1 cutesv=2.1.0 -y
conda create --prefix /lizardfs/guarracino/condatools/wally/0.5.8/ -c conda-forge -c bioconda wally=0.5.8 -y

cd /lizardfs/guarracino/wfmash-paper/scripts
wget -c https://gitlab.com/mcfrith/last/-/raw/main/bin/maf-convert
```

### Alignment

```bash
conda activate /lizardfs/guarracino/condatools/wfmash-paper/

mkdir -p /lizardfs/guarracino/wfmash-paper/alignment/athaliana/
cd /lizardfs/guarracino/wfmash-paper/alignment/athaliana/

SAMPLE=/lizardfs/guarracino/wfmash-paper/assemblies/athaliana/Ler-0.fa
REF=/lizardfs/guarracino/wfmash-paper/assemblies/athaliana/Col-CC.fa
NAME=$(basename $SAMPLE .fa)
REF_NAME=$(basename $REF .fa)

REF_ANNOTATION=/lizardfs/guarracino/wfmash-paper/assemblies/athaliana/annotation/Col-CC.liftoff.gff3
fmt="%C\\n%Us user %Ss system %P cpu %es total %MKb max memory"


## minimap2
for p in asm5 asm20; do
    #\time -f "$fmt" minimap2 -x $p -c --eqx -t 36 $REF $SAMPLE > $NAME.mm2_${p}.paf
    \time -f "$fmt" minimap2 -ax $p -c --eqx -t 36 $REF $SAMPLE > $NAME.mm2_${p}.sam
done


## winnowmap2
#$time -f "$fmt" meryl count k=19 output merylDB asm1.fa
#$time -f "$fmt" meryl print greater-than distinct=0.9998 merylDB > repetitive_k19.txt
#$time -f "$fmt" winnowmap -W repetitive_k19.txt -ax asm20 --eqx -t 20 ./Col-CC.fa $SAMPLE.fa > $SAMPLE.winowmap2_asm20.sam


## AnchorWave
\time -f "$fmt" anchorwave gff2seq -i $REF_ANNOTATION -r $REF -o $REF_NAME.filter.cds.fa
\time -f "$fmt" minimap2 -x splice -t 10 -k 12 -a -p 0.4 -N 20 $REF $REF_NAME.filter.cds.fa > $REF_NAME.cds.sam
\time -f "$fmt" minimap2 -x splice -t 10 -k 12 -a -p 0.4 -N 20 $SAMPLE $REF_NAME.filter.cds.fa > $NAME.cds.sam
## -IV: within-species with inverison
\time -f "$fmt" anchorwave genoAli -i $REF_ANNOTATION -as $REF_NAME.filter.cds.fa -r $REF -a $NAME.cds.sam -ar $REF_NAME.cds.sam -s $SAMPLE -n $NAME.aw.anchors -o $NAME.aw.maf -f $NAME.aw.m.maf -IV
rm $NAME.aw.m.maf
## https://gitlab.com/mcfrith/last/-/blob/main/bin/maf-convert
python /lizardfs/guarracino/wfmash-paper/scripts/maf-convert sam $NAME.aw.maf > $NAME.aw.sam


# mummmer
\time -f "$fmt" nucmer --sam-long $NAME.nucmer.sam --maxmatch -c 500 -b 500 -l 100 -t 8 $REF $SAMPLE
conda deactivate

# wfmash

# Paths for local testing
SAMPLE=/home/guarracino/Desktop/Garrison/wfmash-paper/Ler-0.fa
REF=/home/guarracino/Desktop/Garrison/wfmash-paper/Col-CC.fa
NAME=$(basename $SAMPLE .fa)
REF_NAME=$(basename $REF .fa)

#for p in 80 90 95; do
for p in 95; do
    #\time -f "$fmt" wfmash -s 10000 -p $p -n 1 -t 36 --hg-filter-ani-diff 30 $REF $SAMPLE > $NAME.wfmash_p${p}s10kn1k19.paf
    \time -f "$fmt" wfmash -a -s 10000 -p $p -n 1 -t 36 --hg-filter-ani-diff 30 $REF $SAMPLE > $NAME.wfmash_p${p}s10kn1k19.sam
done
```

## Variant calling

### SVs

```shell
SAMPLE=/lizardfs/guarracino/wfmash-paper/assemblies/athaliana/Ler-0.fa
REF=/lizardfs/guarracino/wfmash-paper/assemblies/athaliana/Col-CC.fa
NAME=$(basename $SAMPLE .fa)
REF_NAME=$(basename $REF .fa)

# Combine SAM files into a sorted BAM file
grep "^@" $NAME.mm2_asm20.sam > header
rm combine.sam
for i in `ls *.sam | grep cds -v |sed -e "s/.sam//g" -e "s/${NAME}.//g"`; do
    echo "$i"
    if [[ $i == *aw* ]]; then
        grep -v "^@" ${NAME}.${i}.sam >> combine.sam
        cat header <(grep -v "^@" ${NAME}.${i}.sam) | samtools sort -@ 24 -T /scratch/$i -O bam -o ${NAME}.${i}.sorted.bam
    elif [[ $i == *wfmash* ]]; then
        grep -v "^@" ${NAME}.${i}.sam | sed -e "s/^Chr/${NAME}#${i}#1#Chr/g" >> combine.sam
        samtools sort -@ 24 -T /scratch/$i -O bam -o ${NAME}.${i}.sorted.bam ${NAME}.${i}.sam
    elif [[ $i == *mm2* ]];then
        grep -v "^@" ${NAME}.${i}.sam | sed -e "s/^Chr/${NAME}#${i}#1#Chr/g" >> combine.sam
        samtools sort -@ 24 -T /scratch/$i -O bam -o ${NAME}.${i}.sorted.bam ${NAME}.${i}.sam
    fi
done
echo "combine"
cat header combine.sam | samtools sort -@ 24 -T /scratch/$i -O bam -o $NAME.aw_wfmash_mm2.sorted.bam
rm combine.sam header
samtools index -@ 24 $NAME.aw_wfmash_mm2.sorted.bam


# cutesv
conda activate /lizardfs/guarracino/condatools/wfmash-paper/
cuteSV $NAME.aw_wfmash_mm2.sorted.bam $REF $NAME.aw_wfmash_mm2.cutesv.vcf ./ \
    -t 48 -s 1 --genotype --report_readid -p -1 -mi 500 -md 500 --min_mapq 0 \
    --max_cluster_bias_INS 1000 \
    --diff_ratio_merging_INS 0.9 \
    --max_cluster_bias_DEL 1000 \
    --diff_ratio_merging_DEL 0.5 \
    --min_size 50
bgzip -@ 48 $NAME.aw_wfmash_mm2.cutesv.vcf -f
conda deactivate

# Filter SVs
bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.vcf.gz | awk '$3 == 1' | grep "wfmash" | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_only.bed
bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.vcf.gz | awk '$3 != 4' | grep "wfmash" -v | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_no.bed

conda activate /lizardfs/guarracino/condatools/wally/0.5.8

wally region -R <(head $NAME.wfmash_only.bed) -g $REF $NAME.aw_wfmash_mm2.sorted.bam
```





# STUFF FOR DEVELOPMENT

```shell
wfmash-convex-patching -a -s 10000 -p $p -n 1 -t 48 --hg-filter-ani-diff 30 $REF $SAMPLE $WFA_PATCH_PARAMS -m > $NAME.wfmash_p${p}s10kn1k19.map

for m in 4 5 6 7; do
    for o1 in 6 7 8 9 10; do
        for o2 in `seq 26 61`; do
            SUFFIX="$m.$o1.2.$o2.1"
            echo $SUFFIX

            mkdir -p $SUFFIX
            cd $SUFFIX

            ##############################################################################################
            p=95; wfmash-convex-patching -a -s 10000 -p $p -n 1 -t 96 --hg-filter-ani-diff 30 $REF $SAMPLE --wfa-patching-params $m,$o1,2,$o2,1 -i ../$NAME.wfmash_p${p}s10kn1k19.map > $NAME.wfmash_p${p}s10kn1k19.sam

            grep "^@" ../$NAME.mm2_asm20.sam > header

            rm combine.sam
            i=aw
            grep -v "^@" ../${NAME}.${i}.sam >> combine.sam
            i=mm2_asm20
            grep -v "^@" ../${NAME}.${i}.sam | sed -e "s/^Chr/${NAME}#${i}#1#Chr/g" >> combine.sam
            i=mm2_asm5
            grep -v "^@" ../${NAME}.${i}.sam | sed -e "s/^Chr/${NAME}#${i}#1#Chr/g" >> combine.sam
            i=wfmash_p95s10kn1k19
            grep -v "^@" ${NAME}.${i}.sam | sed -e "s/^Chr/${NAME}#${i}#1#Chr/g" >> combine.sam
            echo "combine"
            cat header combine.sam | samtools sort -@ 96 -T /scratch/$i -O bam -o $NAME.aw_wfmash_mm2.sorted.bam
            rm combine.sam header
            samtools index -@ 96 $NAME.aw_wfmash_mm2.sorted.bam

            cuteSV $NAME.aw_wfmash_mm2.sorted.bam $REF $NAME.aw_wfmash_mm2.cutesv.vcf ./ \
                -t 96 -s 1 --genotype --report_readid -p -1 -mi 500 -md 500 --min_mapq 0 \
                --max_cluster_bias_INS 1000 \
                --diff_ratio_merging_INS 0.9 \
                --max_cluster_bias_DEL 1000 \
                --diff_ratio_merging_DEL 0.5 \
                --min_size 50
            bgzip -@ 96 $NAME.aw_wfmash_mm2.cutesv.vcf
            bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.vcf.gz | awk '$3 == 1' | grep "wfmash" | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_only.bed

            cuteSV $NAME.aw_wfmash_mm2.sorted.bam $REF $NAME.aw_wfmash_mm2.cutesv.min5.vcf ./ \
                -t 96 -s 1 --genotype --report_readid -p -1 -mi 500 -md 500 --min_mapq 0 \
                --max_cluster_bias_INS 1000 \
                --diff_ratio_merging_INS 0.9 \
                --max_cluster_bias_DEL 1000 \
                --diff_ratio_merging_DEL 0.5 \
                --min_size 5
            bgzip -@ 96 $NAME.aw_wfmash_mm2.cutesv.min5.vcf

            bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.min5.vcf.gz | awk '$3 == 1' | grep "wfmash" | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_only.min5.bed
            #bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.min5.vcf.gz | awk '$3 != 4' | grep "wfmash" -v | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_no.min5.bed

            cuteSV $NAME.aw_wfmash_mm2.sorted.bam $REF $NAME.aw_wfmash_mm2.cutesv.min1.vcf ./ \
                -t 96 -s 1 --genotype --report_readid -p -1 -mi 500 -md 500 --min_mapq 0 \
                --max_cluster_bias_INS 1000 \
                --diff_ratio_merging_INS 0.9 \
                --max_cluster_bias_DEL 1000 \
                --diff_ratio_merging_DEL 0.5 \
                --min_size 1
            bgzip -@ 96 $NAME.aw_wfmash_mm2.cutesv.min1.vcf

            bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.min1.vcf.gz | awk '$3 == 1' | grep "wfmash" | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_only.min1.bed
            #bcftools query -f "%CHROM\t%POS\t%INFO/RE\t%INFO/RNAMES\n" $NAME.aw_wfmash_mm2.cutesv.min1.vcf.gz | awk '$3 != 4' | grep "wfmash" -v | awk '{print $1"\t"$2"\t"$2+1"\t"$4}' > $NAME.wfmash_no.min1.bed
            ##############################################################################################

            rm $NAME.wfmash_p${p}s10kn1k19.sam
            cd ..
        done
    done
done


```

## Comparison
